src/components/common/input/index.tsx

```diff
@@ -0,0 +1,66 @@
+'use client';
+
+import { InputHTMLAttributes, useState } from 'react';
+import clsx from 'clsx';
+
+interface InputOrTextarea {
+  leftSlot?: React.ReactNode;
+  rightSlot?: React.ReactNode;
+  isSuccess?: boolean;
+  isFail?: boolean;
+  className?: string;
+  ref?: React.Ref<HTMLInputElement>;
+}
+
+interface InputProps extends InputHTMLAttributes<HTMLInputElement>, InputOrTextarea {}
+
+export default function Input({
+  leftSlot = null,
+  rightSlot = null,
+  isSuccess,
+  isFail,
+  className,
+  ref,
+  ...rest
+}: InputProps) {
+  const [isBlurred, setIsBlurred] = useState(false);
+  const [isFocused, setIsFocused] = useState(false);
+
+  const handleFocus = (e: React.FocusEvent<HTMLInputElement>) => {
+    setIsFocused(true);
+    rest.onFocus?.(e);
+  };
+
+  const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+    setIsFocused(false);
+    setIsBlurred(true);
+    rest.onBlur?.(e);
+  };
+
+  const showSuccess = isBlurred && isSuccess === true;
+  const showError = isBlurred && isFail === true;
+
+  return (
+    <div
+      className={clsx(
+        'bg-bg200 flex h-11 w-full gap-3 rounded-xl border px-4 py-2.5 sm:h-12',
+        (isFocused || (!showSuccess && !showError)) && 'border-border',
+        !isFocused && showSuccess && 'border-primary',
+        !isFocused && showError && 'border-danger',
```

Comment:
```suggestion
border-border,
isSuccess && "border-primary",
isFail && "border-danger",
```

처럼 작성해도 일단은 될 거 같은데 저도 모든 경우의 수를 테스트해본 건 아니라 확실하진 않습니다!

여담이지만 `isSuccess`와 대응하려면 `isFail` 보다는 `isFailure`가 더 정확할 것 같습니다 😅